README.md
# Race Up Webpage
*You can find the website [here](https://raceup.github.io/)*

## Install
0. Clone this repository
0. install dependencies with scripts/install.sh`
0. run `scripts/run.sh`
0. open a browser [here](http://127.0.0.1:4000/)

## Questions and issues
The [github issue tracker](https://github.com/raceup/raceup.github.io/issues) is **only** for bug reports and feature requests. Anything else, such as questions for help, should be mailed [here](mailto:info@raceup.it).

## License
[Apache License](http://www.apache.org/licenses/LICENSE-2.0) Version 2.0, January 2004


CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

## 0.9.4 - 2017-10-13

### Added
- closing recruiting at 14 oct at 12:00 script

## 0.9.3 - 2017-10-10

### Refactor
- minified css/js/html

## 0.9.2 - 2017-10-09

### Refactor
- unified it/en pages

### Remove
- about page (not used)

## 0.9.1 - 2017-10-04

### Added
- post with image

## 0.9 - 2017-09-29

### Fixed
- header logos

## 0.8.9 - 2017-09-12

### Removed
- json db

### Fixed
- post url for eng version
- recruiting end date

### Added
- eng version

## 0.8.8 - 2017-09-10

### Added
- bitcoin button donations
- press staments

### Removed
- varano post

## 0.8.7 - 2017-09-03

### Added
- members db
- members page creator script
- comments section in news page

### Fixed
- uploader/syncer script

## 0.8.6 - 2017-08-25

### Added
- fsg 2017 media gallery
- varano 2017 media gallery
- script to add images in 'media' section
- script to edit images
- uploader/syncer via ftp
- olds scripts to parse website

### Refactored
- mkdir 'scripts' with tools to automate site creation
- run/upload scripts to 'scripts' folder

## 0.8.5 - 2017-08-20

### Added
- knipex logo with description in ed 2nd class
- 'name' attr in sponsors

### Removed
- duplicate 'mattarollo' sponsor in cd page

## 0.8.4 - 2017-08-15

### Added
- bitbucket ed sponsor in 3rd class

### Fixed
- meder sponsor image

## 0.8.3 - 2017-08-04

### Added
- varano press review post
- meccanicampz url link in sponsors page

### Fixed
- feed items link

## 0.8.2 - 2017-07-25

### Added
- varano victory post

### Fixed
- malformed email link in post sharing
- malformed post permalink when sharing
- duplicate .html ending in post permalinks
- missing http:// in post sharing urls

## 0.8.1 - 2017-07-04

### Added
- nuova tempra cd logo

### Fixed
- duplicate row in cd sponsors
- schroth cd logo

## 0.8 - 2017-06-24

### Added
- zf race camp recap post

## 0.7.9.5 - 2017-06-21

### Added
- meder schroth ed logo

### Fixed
- missing biz image

## 0.7.9.4 - 2017-06-08

### Added
- schroth cd logo

## 0.7.9.3 - 2017-06-06

### Added
- new raw zf images
- zf media section with link to full-size images

### Removed
- academy page

## 0.7.9.2 - 2017-05-28

### Fixed
- background image width
- header with logos on single line

### Added
- ganalytics script

## 0.7.9.1 - 2017-05-28

### Removed
- twitter, bitbucket, linkedin, rss, github footer icons

### Added
- background image with opacity

## 0.7.9 - 2017-05-27

### Added
- new banner
- new units images
- new sponsor logos

## 0.7.8 - 2017-05-27

### Added
- academy page and content
- academy submenu in header
- biz team image
- new ED logo in timeline

### Fixed
- header nav width
- out of focus oz logo

## 0.7.7 - 2017-05-27

### Fixed
- number of students in project description
- removed Thomas from team

### Removed
- Sonic, Bercella, Endrich components sponsor logos

## 0.7.6 - 2017-05-15

### Added
- 2017 media images

## 0.7.5 - 2017-05-07

### Added
- noaloil cd logo

## 0.7.4 - 2017-05-06

### Added
- post about galileo fesitval 2017
- image in galileo2017 post

### Removed
- tm cd sponsor

## 0.7.3 - 2017-04-27

### Added
- post about 2017 cars launch
- fixed twitter post share link

## 0.7.2 - 2017-04-15

### Added
- social icons in footer
- paypal link-on-page
- post about paypal

### Changed
- title in paypal post

### Refactored
- module for paypal button

## 0.7.1 - 2017-04-11

### Added
- 'rossi' logo in cd

### Fixed
- cd logos centering

## 0.7 - 2017-04-02

### Added
- paypal 'donate' button in home and support page
- made with love in footer

## 0.6.9 - 2017-03-30

### Added
- bcomp sponsor in cd-2

## 0.6.8 - 2017-03-22

### Added
- tecnolaser sponsor in cd-2
- post mecspe 4
- media > video mecspe 4

### Fixed
- post mecspe 4

## 0.6.7 - 2017-03-21

### Removed
- calogero, cipriani, berton, zingale from team members

### Changed
- padoan is cd-suspensions manager

## 0.6.6 - 2017-03-19

### Added
- 'mecspe pt3' post
- 'mecspe pt3' video in 'media' section
- 'patrick' as 'RESPONSABILE PRODUZIONE DIVISIONE ELETTRICA'

### Fixed
- page format in 'mecspe3' post

## 0.6.5 - 2017-03-18

### Fixed
- 'fea' english name in brochure

### Added
- cd/ed renderings in team unit
- new logo

## 0.6.5 - 2017-03-18

### Fixed
- 'bleron' school thanks to @Michele
- 'join us' section picture on the left, description on the right
- 'efesto lab' in big sponsors
- 'efesto lab' img

### Added
- 'join us' section with luca
- part of missing cd, ed logos descriptions
- brochure links in home, media and contact us sections

## 0.6.4 - 2017-03-13

### Added
- mecspe2 post
- ballotta video in 'media' section

## 0.6.3 - 2017-03-09

### Fixed
- redirection from raceup.it/electricdivision
- mobile videos in media section too little

### Removed
- 'pastò' from 2017 members

### Added
- 'sorma' to sponsors ed
- mini-gallery with cars images on cars pages

## 0.6.2 - 2017-03-07

### Fixed
- previous/next post link in news

## 0.6.1 - 2017-03-06

### Changed
- 'global informatica' sponsor row

### Added
- sponsors description
- 'stip', 'alth', 'ptc', 'cemes', 'dynamec', 'festo', 'mattarollo' sponsor link

### Fixed
- 'omega', 'cometec', 'tecnocurve', 'protomaker' sponsor link

## 0.6 - 2017-03-05

### Fixed
- url redirect when 404
- redirection to old urls thanks to mori
- optimized (~5%) images
- too large video in mecspe post

### Added
- global informatica sponsor
- mecspe 2017 post
- links to logo images (both ed and cd)
- mecspe video in mecspe post
- old videos in media section
- css `caption` class

## 0.5.9 - 2017-03-05

### Added
- contact us in menu

js/init.js
(function($) {
	skel.init({
		reset: 'full',
		breakpoints: {
			global:		{ range: '*', href: '/css/style.css', containers: 1400, grid: { gutters: 50 } },
			wide:		{ range: '-1680', href: '/css/style-wide.css', containers: 1200, grid: { gutters: 40 } },
			normal:		{ range: '-1280', href: '/css/style-normal.css', containers: 960, viewport: { scalable: false } },
			narrow:		{ range: '-980', href: '/css/style-narrow.css', containers: '90%', grid: { gutters: 30 } },
			narrower:	{ range: '-960', href: '/css/style-narrower.css', grid: { collapse: 1 } },
			mobile:		{ range: '-736', href: '/css/style-mobile.css', containers: '100%', grid: { gutters: 15, collapse: 2 } }
		},
		plugins: {
			layers: {
				config: {
					transformTest: function() { return (skel.vars.isMobile); }
				},
				topPanel: {
					states: '/global/wide/normal/narrow/narrower/mobile',
					position: 'top-center',
					side: 'top',
					hidden: true,
					animation: 'pushY',
					width: '100%',
					height: 275,
					html: '<nav data-action="navList" data-args="nav"></nav>',
					clickToHide: true,
					swipeToHide: false,
					orientation: 'vertical'
				},
				topButton: {
					states: '/global/wide/normal/narrow/narrower/mobile',
					position: 'top-center',
					width: 120,
					height: 50,
					html: '<span class="toggle" data-action="toggleLayer" data-args="topPanel"></span>'
				},
				sidePanel: {
					states: '/global/wide/normal/narrow/narrower',
					position: 'top-left',
					side: 'left',
					hidden: true,
					animation: 'revealX',
					width: 250,
					height: '100%',
					html: '<nav data-action="navList" data-args="nav"></nav>',
					clickToHide: true,
					orientation: 'vertical'
				},
				sideButton: {
					states: '/global/wide/normal/narrow/narrower',
					position: 'top-left',
					width: 100,
					height: 60,
					html: '<span class="toggle" data-action="toggleLayer" data-args="sidePanel"></span>'
				}
			}
		}
	});

	$(function() {

		var	$window = $(window),
			$body = $('body'),
			$header = $('#header'),
			$banner = $('#banner');

		// Disable animations/transitions until the page has loaded.
			$body.addClass('is-loading');
			
			$window.on('load', function() {
				$body.removeClass('is-loading');
			});

		// CSS polyfills (IE<9).
			if (skel.vars.IEVersion < 9)
				$(':last-child').addClass('last-child');

		// Forms (IE<10).
			var $form = $('form');
			if ($form.length > 0) {

				if (skel.vars.IEVersion < 10) {
					$.fn.n33_formerize=function(){var _fakes=new Array(),_form = $(this);_form.find('input[type=text],textarea').each(function() { var e = $(this); if (e.val() == '' || e.val() == e.attr('placeholder')) { e.addClass('formerize-placeholder'); e.val(e.attr('placeholder')); } }).blur(function() { var e = $(this); if (e.attr('name').match(/_fakeformerizefield$/)) return; if (e.val() == '') { e.addClass('formerize-placeholder'); e.val(e.attr('placeholder')); } }).focus(function() { var e = $(this); if (e.attr('name').match(/_fakeformerizefield$/)) return; if (e.val() == e.attr('placeholder')) { e.removeClass('formerize-placeholder'); e.val(''); } }); _form.find('input[type=password]').each(function() { var e = $(this); var x = $($('<div>').append(e.clone()).remove().html().replace(/type="password"/i, 'type="text"').replace(/type=password/i, 'type=text')); if (e.attr('id') != '') x.attr('id', e.attr('id') + '_fakeformerizefield'); if (e.attr('name') != '') x.attr('name', e.attr('name') + '_fakeformerizefield'); x.addClass('formerize-placeholder').val(x.attr('placeholder')).insertAfter(e); if (e.val() == '') e.hide(); else x.hide(); e.blur(function(event) { event.preventDefault(); var e = $(this); var x = e.parent().find('input[name=' + e.attr('name') + '_fakeformerizefield]'); if (e.val() == '') { e.hide(); x.show(); } }); x.focus(function(event) { event.preventDefault(); var x = $(this); var e = x.parent().find('input[name=' + x.attr('name').replace('_fakeformerizefield', '') + ']'); x.hide(); e.show().focus(); }); x.keypress(function(event) { event.preventDefault(); x.val(''); }); });  _form.submit(function() { $(this).find('input[type=text],input[type=password],textarea').each(function(event) { var e = $(this); if (e.attr('name').match(/_fakeformerizefield$/)) e.attr('name', ''); if (e.val() == e.attr('placeholder')) { e.removeClass('formerize-placeholder'); e.val(''); } }); }).bind("reset", function(event) { event.preventDefault(); $(this).find('select').val($('option:first').val()); $(this).find('input,textarea').each(function() { var e = $(this); var x; e.removeClass('formerize-placeholder'); switch (this.type) { case 'submit': case 'reset': break; case 'password': e.val(e.attr('defaultValue')); x = e.parent().find('input[name=' + e.attr('name') + '_fakeformerizefield]'); if (e.val() == '') { e.hide(); x.show(); } else { e.show(); x.hide(); } break; case 'checkbox': case 'radio': e.attr('checked', e.attr('defaultValue')); break; case 'text': case 'textarea': e.val(e.attr('defaultValue')); if (e.val() == '') { e.addClass('formerize-placeholder'); e.val(e.attr('placeholder')); } break; default: e.val(e.attr('defaultValue')); break; } }); window.setTimeout(function() { for (x in _fakes) _fakes[x].trigger('formerize_sync'); }, 10); }); return _form; };
					$form.n33_formerize();
				}

			}
			
		// Scrolly links.
			$('.scrolly').scrolly(1000, -10);

		// Dropdowns.
			$('#nav > ul').dropotron({
				mode: 'fade',
				noOpenerFade: true,
				expandMode: (skel.vars.isTouch ? 'click' : 'hover')
			});

		// Header.
		// If the header is using "alt" styling and #banner is present, use scrollwatch
		// to revert it back to normal styling once the user scrolls past the banner.
		// Note: This is disabled on mobile devices.
			if (!skel.vars.isMobile
			&&	$header.hasClass('alt')
			&&	$banner.length > 0) {

				$window.on('load', function() {

					$banner.scrollwatch({
						delay:		0,
						range:		1,
						anchor:		'top',
						on:			function() { $header.addClass('alt reveal'); },
						off:		function() { $header.removeClass('alt'); }
					});

				});
			
			}

	});
})(jQuery);

/* Redirect in case of missing team member image */
function imgAccountError(image) {
    image.onerror = "";
    image.src = "/images/team/account.png";
    image.parentElement.className = "image";
    return true;
}